www/js/app.js

```diff
@@ -998,16 +1001,34 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
             document.getElementById('articleContent').contentWindow.focus();
             $("#searchingArticles").show();
             if (dirEntry.isRedirect()) {
-                selectedArchive.resolveRedirect(dirEntry, readArticle);
+                // Update expectedArticleURLToBeDisplayed before redirect. 
+                selectedArchive.resolveRedirect(dirEntry, function (resolvedDirEntry) {
+                    expectedArticleURLToBeDisplayed = resolvedDirEntry.namespace + "/" + resolvedDirEntry.url;
+                    readArticle(resolvedDirEntry);
+                });
             } else {
                 params.isLandingPage = false;
+                expectedArticleURLToBeDisplayed  = dirEntry.namespace + "/" + dirEntry.url;
                 readArticle(dirEntry);
             }
         } else {
             alert("Data files not set");
         }
     }
 
+    /**
+     * Check whether the given URL from given dirEntry equals the expectedArticleURLToBeDisplayed
+     * @param {DirEntry} dirEntry The directory entry of the article to read
+     */
+    function isDirEntryExpectedToBeDisplayed(dirEntry) {
+        var curArticleURL = dirEntry.namespace + "/" + dirEntry.url;
+
+        if (expectedArticleURLToBeDisplayed !== curArticleURL) {
+            return false;
```

Comment:
I think it's necessary to keep a debug log in this case.
So that, when testing the app, we can see that there has been an early return